### ISSUE TITLE
Inplace operation error raised on `loss.backward()` call

### ISSUE THREAD

**Issue by shrivastava95** [shrivastava95/clip-ocr#3]
This error is being raised due to the loss.backward() call. Swapping `encode_text_coop` with `encode_text` causes the error to no longer be raised. Figure out the error in the implementation or modify it so that it stops raising this error.

![image](https://github.com/shrivastava95/clip-ocr/assets/75275497/c27fb01e-fd1d-4efc-9392-290c49d8fcb6)
